controllers/semeru_compiler.go

```diff
@@ -0,0 +1,150 @@
+/*
+  Copyright contributors to the WASdev project.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+package controllers
+
+import (
+	wlv1 "github.com/WASdev/websphere-liberty-operator/api/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	SemeruLabelNameSuffix = "-semeru-compiler"
+	SemeruLabelName       = "semeru-compiler"
+	JitServer             = "jitserver"
+	ImageURL              = "docker.io/ibm-semeru-runtimes:open-17.0.3_7-jre"
+)
+
+// Create the Deployment and Service objects for a Semeru Compiler used by a Websphere Liberty Application
+func (r *ReconcileWebSphereLiberty) reconcileSemeruCompiler(wlva *wlv1.WebSphereLibertyApplication) (error, string) {
+	compilerMeta := metav1.ObjectMeta{
+		Name:      wlva.GetName() + SemeruLabelNameSuffix,
+		Namespace: wlva.GetNamespace(),
+	}
+	//Deployment
+	semeruDeployment := &appsv1.Deployment{ObjectMeta: compilerMeta}
+	err := r.CreateOrUpdate(semeruDeployment, wlva, func() error {
+		r.reconcileSemeruDeployment(wlva, semeruDeployment)
+		return nil
+	})
+	if err != nil {
+		return err, "Failed to reconcile Deployment : " + semeruDeployment.Name
+	}
+	// Create the Semeru Service object
+	semsvc := &corev1.Service{ObjectMeta: compilerMeta}
+	err = r.CreateOrUpdate(semsvc, wlva, func() error {
+		reconcileSemeruService(semsvc, wlva)
+		return nil
+	})
+	if err != nil {
+		return err, "Failed to reconcile the Semeru Compiler Service"
+	}
+	return nil, ""
+}
+
+func (r *ReconcileWebSphereLiberty) reconcileSemeruDeployment(wlva *wlv1.WebSphereLibertyApplication, deploy *appsv1.Deployment) {
+
+	deploy.Labels = getLabels(wlva)
+	deploy.Spec.Strategy.Type = appsv1.RecreateDeploymentStrategyType
+	replicas := int32(1)
+	deploy.Spec.Replicas = &replicas
+
+	if deploy.Spec.Selector == nil {
+		deploy.Spec.Selector = &metav1.LabelSelector{
+			MatchLabels: getSelectors(wlva),
+		}
+	}
+
+	deploy.Spec.Template = corev1.PodTemplateSpec{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   wlva.Name + SemeruLabelName,
+			Labels: getLabels(wlva),
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:            JitServer,
+					Image:           ImageURL,
+					ImagePullPolicy: "IfNotPresent",
```

Comment:
Use the same pull policy as application since the same application image is used